packages/web-components/src/horizontalbarchart/horizontalbarchart.ts

```diff
@@ -220,6 +243,61 @@ export class HorizontalBarChart extends FASTElement {
           nodes[i].setAttribute(key, attributes[key]);
         });
       });
+
+    const legendContainer = document.createElement('div');
+    div.node()!.appendChild(legendContainer);
+    legendContainer.classList.add('legendcontainer');
+
+    this.uniqueLegends?.forEach(d => {
+      const button = document.createElement('button');
+      legendContainer.appendChild(button);
+      button.classList.add('legend');
+
+      const legendRect = document.createElement('div');
+      button.appendChild(legendRect);
+      legendRect.classList.add('legendRect');
+      legendRect.style['backgroundColor'] = d.color!;
+      legendRect.style['borderColor'] = d.color!;
+
+      const legendText = document.createElement('div');
+      button.appendChild(legendText);
+      legendText.textContent = d.legend!;
+      legendText.classList.add('legendText');
+    });
+
+    const buttons = legendContainer.getElementsByTagName('button');
+    const bars = this.shadowRoot?.querySelectorAll('.bar');
+
+    for (let i = 0; i < buttons.length; i++) {
+      buttons[i].addEventListener('mouseover', () => {
+        for (let j = 0; j < bars!.length; j++) {
+          if (bars![j].getAttribute('barinfo') !== buttons[i].textContent) {
+            bars![j].style['opacity'] = '0.1';
+          }
+        }
+        for (let j = 0; j < buttons.length; j++) {
+          if (j !== i) {
+            const legendRect = buttons[j].getElementsByClassName('legendRect')[0];
+            legendRect.style['backgroundColor'] = 'transparent';
```

Comment:
instead of setting the style manually, could you put the style in a class and assign the correct class or subclass

Reply:
Okay